opencontainers/struct.py

```diff
@@ -63,42 +79,36 @@ def _is_struct(self, attType=None):
         # We can provide a nested attType to check
         if not attType:
             attType = self.attType
-        try:
-            return (
-                Struct in attType.__bases__
-                or StrStruct in attType.__bases__
-                or IntStruct in attType.__bases__
-            )
-        except:
-            return False
+        return is_struct(attType)
 
     def set(self, value):
         """
         Set a new value, and validate the type. Return true if set
         """
 
-        # First pass, it might be another object to add
-        if self._is_struct():
-            newStruct = self.attType()
-            value = newStruct.load(value)
-
-        # If it's a list with another type
-        elif isinstance(self.attType, list) and self.attType:
-            child = self.attType[0]
-
-            # It's either a nested structure
-            if self._is_struct(child):
-
-                # If we have a list of values, generate them
-                if isinstance(value, list):
-                    values = []
-                    for v in value:
+        if not is_struct(type(value)):
```

Comment:
Huh, I think in my mind to "add" means to append or extend a structure. This looks like it would take an int, for example, and replace with another int. This (to me) feels more like an update - is there any reason you feel strongly that it belongs in add?

Reply:
Didn't know your original use case of `.add()`, but makes sense.
But when `.add()` should only append or update a list or dict, we need a `.set()` in `Struct` to allow setting a value directly instead of accessing the `Struct.attrs['whatever']` directly?!

Reply:
Sure that would work!

Reply:
I've now added a `set(name, value)` to set a attributes value.
`add()` is now only allowed on lists or dicts.
I know that this is a breaking change...

Reply:
Why is it a breaking change?

Reply:
Do you feel strongly that we should have one function (add) for adding any type?

Reply:
> Why is it a breaking change?
Because someone might have used `add()` to set the value of an attribute which is not of type list or dict.

> Do you feel strongly that we should have one function (add) for adding any type?
I've just continued what you've started in #13.
And imho I would stick to one function for both to keep it simple.
Don't see a reason to split that up.

Reply:
I think I’ve changed my mind about it too - sorry for the run around!